### PR TITLE
Implemented the API for releasing rulesets

### PR DESCRIPTION
**Pull request by hiranya911**
Implemented the `releaseFirestoreRuleset()` API. This attempts to PATCH the existing release.

### CODE REVIEW

**Comment by ryanpbrewster** [comment]
Under what circumstances would we expect to get a `not-found` error? In general the partner services themselves are responsible for creating the release.

**Comment by hiranya911** [comment]
I'm not really sure -- just going by intuition here. If these releases are guaranteed to always exist, we can certainly get rid of the createRelease logic and further simplify the implementation. There is also a deleteRelease operation supported on the REST API, which I haven't played with. What happens if somebody deletes a release (perhaps accidentally)?

**Comment by ryanpbrewster** [comment]
My mental model is that creating a release is basically "enabling" the service (e.g., linking a GCS bucket up to work with Firebase Storage), and deleting the release is the inverse (e.g., removing the bucket, disabling client access).

**Comment by hiranya911** [comment]
Fair enough. I've removed the createRelease functionality from the PR.

File: src/security-rules/security-rules-api-client.ts [hunk 1]
```diff
@@ -24,8 +24,8 @@ const RULES_V1_API = 'https://firebaserules.googleapis.com/v1';
 export interface Release {
   readonly name: string;
   readonly rulesetName: string;
-  readonly createTime: string;
-  readonly updateTime: string;
+  readonly createTime?: string;
+  readonly updateTime?: string;
```
Comment (rachelmyers) [inline src/security-rules/security-rules-api-client.ts]: Is there a reason to expect that we won't get the `create` and `updateTime`s back, or you're just signaling that we don't rely on them for anything?

File: src/security-rules/security-rules.ts [hunk 2]
```diff
@@ -115,6 +115,18 @@ export class SecurityRules implements FirebaseServiceInterface {
     return this.getRulesetForRelease(SecurityRules.CLOUD_FIRESTORE);
   }
 
+  /**
+   * Releases the specified ruleset as the current Cloud Firestore ruleset. This makes the specified ruleset
+   * the currently applied ruleset for Cloud Firestore.
```
Comment (rachelmyers) [inline src/security-rules/security-rules.ts]: Total nit; seems like the second sentence is more complete and could stand on its own here.

File: src/security-rules/security-rules.ts [hunk 3]
```diff
@@ -188,6 +200,21 @@ export class SecurityRules implements FirebaseServiceInterface {
         return this.getRuleset(stripProjectIdPrefix(rulesetName));
       });
   }
+
+  private releaseRuleset(ruleset: string | RulesetMetadata, releaseName: string): Promise<void> {
+    if (!validator.isNonEmptyString(ruleset) &&
+      (!validator.isNonNullObject(ruleset) || !validator.isNonEmptyString(ruleset.name))) {
```
Comment (rachelmyers) [inline src/security-rules/security-rules.ts]: It's interesting that the name could come in either as a String or as part of the RulesetMetadata Object. Why did you opt for this and not the simpler signature of only passing in the name as a String, making it the caller's job to pluck out the name if it had an Object?

File: src/security-rules/security-rules-api-client.ts (hunk 1 above)
Comment (hiranya911) [reply src/security-rules/security-rules-api-client.ts]: I use the same interface as an input argument when updating/patching a release. In this case I wanted the flexibility to not specify these 2 attributes. And you're right -- we also don't rely on them for anything else.

File: src/security-rules/security-rules.ts (hunk 3 above)
Comment (hiranya911) [reply src/security-rules/security-rules.ts]: We can use the same private method for releasing storage rulesets. By doing it here we save a few lines from being duplicated.

File: src/security-rules/security-rules.ts (hunk 2 above)
Comment (hiranya911) [reply src/security-rules/security-rules.ts]: Done

**Pull request merged**